src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,181 @@
+package frc.robot.leds;
+
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class LedStrip extends SubsystemBase {
+    private final AddressableLED ledStrip;
+    private AddressableLEDBuffer ledBuffer;
+    private int stripLength = 0;
+    private int offset = 0;
+    private int percentage = 100;
+
+    private double blinkTime = 1;
+    private double fadeDuration = 1;
+
+    private Color primary = new Color();
+    private Color secondary = new Color();
+    private Color currentColor = primary;
+    private Color fadeColor = primary;
+
+    private final Timer timer = new Timer();
+    private LedMode mode = LedMode.SOLID;
+
+    public LedStrip(int port, int length) {
+        ledStrip = new AddressableLED(port);
+        ledBuffer = new AddressableLEDBuffer(length);
+        stripLength = length;
+
+        ledStrip.setLength(length);
+        ledStrip.setData(ledBuffer);
+        ledStrip.start();
+
+        timer.start();
+        timer.reset();
+    }
+
+    public void setLength(int length) {
+        ledBuffer = new AddressableLEDBuffer(length);
+        ledStrip.setLength(length);
+        stripLength = length;
+    }
+
+    public void setState(LedState state){
+        setBlinkTime(state.getBlinkTime());
+        setFadeDuration(state.getFadeDuration());
+        setMode(state.getMode());
+        setPrimary(state.getPrimary());
+        setSecondary(state.getSecondary());
+    }
+
+    /**
+     * Sets the primary color of the LEDs.
+     * It will be used when in the solid mode and as the
+     * initial color in the blink and fade modes
+     * @param color
+     */
+    private void setPrimary(Color color) {
+        primary = color;
+    }
+
+    /**
+     * Sets the secondary color of the LEDs.
+     * It will be used as the second color in the blink and fade modes
+     * @param color
+     */
+    private void setSecondary(Color color) {
+        secondary = color;
+    }
```

Comment:
I'd delete these methods, these members are private so you don't setters for them. You can move the documentation directly above the members' declerations.